src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight2B2T.kt

```diff
@@ -0,0 +1,346 @@
+         * onGround == true (which we obviously need to revert before sending to the server)
+         */
+        safeListener<PacketEvent.Send> {
+            if (started != MovementState.NOT_STARTED) {
+                when (it.packet) {
+                    is CPacketPlayer.Position -> {
+                        if (it.packet.onGround) {
+                            if (player.posY - getGroundPos().y > 0.0) {
+                                it.packet.onGround = false
+                            }
+                        } else {
+                            it.cancel()
+                        }
+                    }
+                    is CPacketPlayer.PositionRotation -> {
+                        if (it.packet.onGround) {
+                            if (player.posY - getGroundPos().y > 0.0) {
+                                it.packet.onGround = false
+                            }
+                        } else {
+                            rotation = Vec2f(it.packet.pitch, it.packet.yaw)
+                            it.cancel()
+                        }
+                    }
+                    is CPacketPlayer.Rotation -> {
+                        rotation = Vec2f(it.packet.pitch, it.packet.yaw)
+                        it.cancel()
+                    }
+                }
+            }
+        }
+
+        /**
+         * Listen for CPacketConfirmTeleport. We must send a PositionRotation immediately.
+         */
+        safeListener<PacketEvent.PostSend> {
+            when (it.packet) {
+                is CPacketConfirmTeleport -> {
+                    if (showDebug.value) sendChatMessage("Responding to emergency teleport packet from the server.")
+                    player.setVelocity(0.0, 0.0, 0.0)
+                    accelStart = System.currentTimeMillis()
+                    /* This only sets the position and rotation client side since it is not salted with onGround */
+                    player.setPositionAndRotation(teleportPosition.x,
+                        teleportPosition.y,
+                        teleportPosition.z,
+                        teleportRotation.y,
+                        teleportRotation.x)
+                    /* Force send the packet */
+                    sendForcedPacket(true)
+                }
+            }
+        }
+
+        BackgroundScope.launchLooping("FlyConnection", 150L) {
+            if (started != MovementState.NOT_STARTED) {
+                onMainThreadSafe {
+                    connection.sendPacket(CPacketEntityAction(player, CPacketEntityAction.Action.START_FALL_FLYING))
+
+                    rotation = player.pitchYaw
+                    sendForcedPacket(false)
+
+                    connection.sendPacket(CPacketEntityAction(player, CPacketEntityAction.Action.START_FALL_FLYING))
+                    connection.sendPacket(CPacketEntityAction(player, CPacketEntityAction.Action.START_FALL_FLYING))
+                }
+            }
```

Comment:
Run this on tick instead of making it a background job

Reply:
That did not work somehow. This was a workaround

Reply:
used tickTimer. It went out of the 150ms cycle easiely

Reply:
> used tickTimer. It went out of the 150ms cycle easiely

What about counter up to 3 on tick?